### PR TITLE
Add task to the list view and a constraint on this field

### PR DESCRIPTION
**Pull request by yasmineelmrini**

### CODE REVIEW

File: project_invoicing/models/project_project.py [hunk 1]
```diff
@@ -78,7 +78,8 @@ def generate_invoices(self, data):
                 'company_id': self.company_id.id,
                 'account_id': partner.property_account_receivable_id.id,
                 'invoice_line_ids': [(0, 0, l) for l in lines],
-                'source_analytic_line_ids': [(6, 0, aal_ids)]
+                'source_analytic_line_ids': [(6, 0, aal_ids)],
+                'is_project_invoice': True,
```
Comment (dufresnedavid) [inline project_invoicing/models/project_project.py]: is_project_invoice should be only in equipment_project.

File: project_invoicing/demo/project_project.xml [hunk 2]
```diff
@@ -1,10 +1,16 @@
 <?xml version="1.0"?>
 <openerp>
-    <data noupdate="1">
+    <data>
+    	
+    	<record model="account.analytic.account" id="account_analytic_account_project">
+    		<field name="name">Analytic account for the Website project</field>
```
Comment (ghost) [inline project_invoicing/demo/project_project.xml]: indentation

**Comment by ghost** [comment]
2 minor comments + régler le conflit and then we can merge.

File: project_invoicing/demo/project_project.xml [hunk 3]
```diff
@@ -2,14 +2,14 @@
...
-    	</record>
+        <record model="account.analytic.account" id="account_analytic_account_project">
+            <field name="name">Analytic account for the Website project</field>
+        </record>
 
         <record model="project.project" id="project_project_website" >
             <field name="name">Website for Accounting</field>
             <field name="user_id" ref="res_users_jack"/>
-            <field name="use_tasks" eval="True"/>
+            <field name="user_tasks" eval="True"/>
```
Comment (ghost) [inline project_invoicing/demo/project_project.xml]: le nom du champ est vraiment use_tasks

File: project_invoicing/models/account_invoice_line.py [hunk 4]
```diff
@@ -2,11 +2,33 @@
...
+            invoice_projects = line.account_analytic_id.project_ids
+            task_project = line.task_id.project_id
+
+            if (
+                task_project and invoice_projects and
+                task_project not in invoice_projects
+            ):
+                raise UserError(_(
+                    'The invoice line (%(line)s) has a task '
+                    '(%(task)s) and the account invoice has a project '
```
Comment (ghost) [inline project_invoicing/models/account_invoice_line.py]: this should be `and the invoice` instead of `and the account invoice`

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**